cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -0,0 +1,148 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.preview;
+
+import co.cask.cdap.api.artifact.ArtifactScope;
+import co.cask.cdap.api.metrics.MetricTimeSeries;
+import co.cask.cdap.app.preview.PreviewManager;
+import co.cask.cdap.app.preview.PreviewStatus;
+import co.cask.cdap.app.runtime.ProgramController;
+import co.cask.cdap.app.runtime.ProgramRuntimeService;
+import co.cask.cdap.common.NotFoundException;
+import co.cask.cdap.common.app.RunIds;
+import co.cask.cdap.internal.app.deploy.ProgramTerminator;
+import co.cask.cdap.internal.app.runtime.AbstractListener;
+import co.cask.cdap.internal.app.services.ApplicationLifecycleService;
+import co.cask.cdap.internal.app.services.ProgramLifecycleService;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.artifact.AppRequest;
+import co.cask.cdap.proto.artifact.ArtifactSummary;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.PreviewId;
+import co.cask.cdap.proto.id.ProgramId;
+import com.google.common.base.Throwables;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import org.apache.twill.api.logging.LogEntry;
+import org.apache.twill.common.Threads;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.Nullable;
+
+/**
+ * Default implementation of the {@link PreviewManager}.
+ */
+public class DefaultPreviewManager implements PreviewManager {
+  private static final Gson GSON = new Gson();
+  private final ApplicationLifecycleService applicationLifecycleService;
+  private final ProgramLifecycleService programLifecycleService;
+  private final Map<PreviewId, PreviewStatus> previewStatusMap = new ConcurrentHashMap<>();
+
+  @Inject
+  DefaultPreviewManager(ApplicationLifecycleService applicationLifecycleService,
+                        ProgramLifecycleService programLifecycleService) {
+    this.applicationLifecycleService = applicationLifecycleService;
+    this.programLifecycleService = programLifecycleService;
+  }
+
+  @Override
+  public PreviewId start(NamespaceId namespaceId, String config) throws Exception {
+    final PreviewId previewId = new PreviewId(namespaceId.getNamespace(), RunIds.generate().getId());
+    AppRequest<?> appRequest = GSON.fromJson(config, AppRequest.class);
+    ArtifactSummary artifactSummary = appRequest.getArtifact();
+    Id.Namespace artifactNamespace =
```

Comment:
we cann `applicationLifecycleService.deployApp` which still uses old  `Id.Artifact` and so does the ArtifactRepository's `getArtifact` they have to be updated to use the new `ArtifactId`... we should do that in a separate PR.